src/fastfield/alive_bitset.rs

```diff
@@ -0,0 +1,200 @@
+use crate::directory::FileSlice;
+use crate::directory::OwnedBytes;
+use crate::space_usage::ByteCount;
+use crate::DocId;
+use common::BitSet;
+use std::io;
+use std::io::Write;
+
+/// Write a alive `BitSet`
+///
+/// where `alive_bitset` is the set of alive `DocId`.
+/// Warning: this function does not call terminate. The caller is in charge of
+/// closing the writer properly.
+pub fn write_alive_bitset<T: Write>(alive_bitset: &BitSet, writer: &mut T) -> io::Result<()> {
+    alive_bitset.serialize(writer)?;
+    Ok(())
+}
+
+/// Set of alive `DocId`s.
+#[derive(Clone)]
+pub struct AliveBitSet {
+    data: OwnedBytes,
+    num_deleted: usize,
+}
+
+impl AliveBitSet {
+    #[cfg(test)]
+    pub(crate) fn for_test(not_alive_docs: &[DocId], max_doc: u32) -> AliveBitSet {
+        use crate::directory::{Directory, RamDirectory, TerminatingWrite};
+        use std::path::Path;
+        assert!(not_alive_docs.iter().all(|&doc| doc < max_doc));
+        let mut bitset = BitSet::with_max_value_and_filled(max_doc);
+        for &doc in not_alive_docs {
+            bitset.remove(doc);
+        }
+        let directory = RamDirectory::create();
+        let path = Path::new("dummydeletebitset");
+        let mut wrt = directory.open_write(path).unwrap();
+        write_alive_bitset(&bitset, &mut wrt).unwrap();
+        wrt.terminate().unwrap();
+        let file = directory.open_read(path).unwrap();
+        Self::open(file).unwrap()
+    }
+
+    /// Opens a delete bitset given its file.
+    pub fn open(file: FileSlice) -> crate::Result<AliveBitSet> {
+        let bytes = file.read_bytes()?;
```

Comment:
What happened to the 4 byte with max_doc in the beginning?

OwnedBytes is sliceable for free.

Reply:
they are still there, but now private data of the BitSet, since it's part of the serialization format of BitSet